src/main/webapp/destinationCreation.js

```diff
@@ -22,31 +32,87 @@ function addScriptToHead() { // eslint-disable-line
 }
 
 function searchForPlace() { // eslint-disable-line
+  const element = document.querySelectorAll('#place');
+  if (element.length > 0) {
+    let i;
+    for (i = 0; i < element.length; i++) {
+      removeElement(element[i].id);
+    }
+  }
   // Coresponds to the location of the Googleplex building
   const mapCenter = new google.maps.LatLng(37.421949, -122.083972);
 
-  const map = new google.maps.Map(document.getElementById('map'), {
+  const map = new google.maps.Map(document.getElementById(MAP), {
     center: mapCenter,
     zoom: 18,
   });
 
   const placeService = new google.maps.places.PlacesService(map);
 
-  const text = document.getElementById('search').value;
+  const text = document.getElementById(SEARCH).value;
 
   const request = {
     query: text,
-    fields: ['name', 'geometry'],
+    fields: ['name', 'geometry', 'place_id'],
   };
 
   placeService.findPlaceFromQuery(request, (results, status) => {
+    let div;
     if (status === google.maps.places.PlacesServiceStatus.OK) {
-      results.forEach((item) => {
-        // TODO: Create clickable divs for each item that is
-        // returned from the query and fill in input fields with appropriate
-        // information provided by the places object
-        console.log(item);
+      results.forEach((place) => {
+        div = document.createElement('div');
+        div.setAttribute('data-place-id', place.place_id);
+        div.innerText = place.name;
+        div.id = 'place';
+        div.onclick = fillInValues;
+        document.getElementsByClassName(SEARCH_RESULTS)[0].appendChild(div);
       });
     }
   });
 }
+
+function fillInValues() { // eslint-disable-line
+  const place = this.dataset.placeId;
+  const nameField = document.getElementById(NAME_INPUT);
+  const latField = document.getElementById(LAT_INPUT);
+  const lngField = document.getElementById(LNG_INPUT);
+  const placeField = document.getElementById('place-input');
+  const mapCenter = new google.maps.LatLng(37.421949, -122.083972);
+  const map = new google.maps.Map(document.getElementById(MAP), {
+    center: mapCenter,
+    zoom: 15,
+  });
+
+  const placeService = new google.maps.places.PlacesService(map);
+  const request = {
+    placeId: place,
+    fields: ['name', 'geometry'],
+  };
+  let marker;
+  const element = document.querySelectorAll('#place');
+  let i;
+  for (i = 0; i < element.length; i++) {
+    removeElement(element[i].id);
+  }
+  placeService.getDetails(request, (result, status) => {
```

Comment:
Do you need to make a second request here?  it seems like the first findPlaceFromQuery should return the geometry info.  Instead of making a second request to get this data again, you should pass this data to the onclick handler.

Tim might have a better way of doing this, but you can use [bind](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) to bind the arguments to the method, like this: https://jsbin.com/midugowidi/edit?html,js,output